.github/workflows/publish-python-client.yml

```diff
@@ -13,10 +13,16 @@ jobs:
       - uses: actions/checkout@v3
         with:
           ref: ${{ github.event.client_payload.ref }}
+          fetch-depth: 0
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: 3.9
+      - name: Generate version from tag
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        run: |
+          tag_version=$(echo ${{ github.ref }} | grep -oP '(?<=refs/tags/v)\d+\.\d+\.\d+'))
```

Comment:
I think it's ok, but this will fail if somebody e.g. uses new version '3.0', right? It might be slightly unexpected, but the requirement could be imposed I guess.

Reply:
I added a validation step, that a valid semantic version string is extracted: https://github.com/ITISFoundation/osparc-simcore-clients/pull/83/commits/18e9776b98c235f190cb3f988d9d8288e404b5e6